Source/WebCore/dom/Document.cpp

```diff
@@ -9787,6 +9789,42 @@ void Document::updateContentRelevancyForScrollIfNeeded(const Element& scrollAnch
     return m_contentVisibilityDocumentState->updateContentRelevancyForScrollIfNeeded(scrollAnchor);
 }
 
+RefPtr<ViewTransition> Document::activeViewTransition() const
+{
+    return m_activeViewTransition;
+}
+
+void Document::setActiveViewTransition(RefPtr<ViewTransition> viewTransition)
+{
+    m_activeViewTransition = WTFMove(viewTransition);
+}
+
+Ref<ViewTransition> Document::startViewTransition(RefPtr<ViewTransitionUpdateCallback> updateCallback)
+{
+    Ref viewTransition = ViewTransition::create(*this, WTFMove(updateCallback));
+
+    if (m_activeViewTransition)
```

Comment:
```
if (RefPtr activeViewTransition = m_activeViewTransition)
    activeViewTransition->skipViewTransition(Exception { ExceptionCode::AbortError, "Old view transition aborted by new view transition."_s });
```

This is our modern smart pointer practice. You need to make sure you have a smart pointer *on the stack* when you call a non-trivial member function on it or pass it as argument to a function.